out_s3.go

```diff
@@ -320,19 +332,27 @@ func FLBPluginFlushCtx(ctx, data unsafe.Pointer, length C.int, tag *C.char) int
 }
 
 // format is S3_PREFIX/S3_TRAILING_PREFIX/date/hour/timestamp_uuid.log
-func GenerateObjectKey(s3operator *s3operator, t time.Time) string {
+func GenerateObjectKey(s3operator *s3operator, t time.Time, lines string) string {
 	var fileext string
 	switch s3operator.compressFormat {
 	case plainTextFormat:
 		fileext = ".log"
 	case gzipFormat:
 		fileext = ".log.gz"
 	}
+	var suffix string
+	switch s3operator.suffixAlgorithm {
+	case noSuffixAlgorithm:
+		suffix = ""
+	case sha256SuffixAlgorithm:
+		b := sha256.Sum256([]byte(lines))
```

Comment:
A collision occurs when the same log appears at the same time.
Such cases are very rare, so I think they are acceptable.